### DISCUSSION TITLE
What is the best way to handle optional authentication? (Resources that may or may not be protected)

### DISCUSSION THREAD

**Discussion by NexRX** [poem-web/poem#629] [Q&A] [answered]
Hi again,

I have a resources that is optionally protected, think a blog post that a user can make private by setting a password.

If the resource is protected, great, we can demand a API key and use that as a password.
If the resource isn't protected, unforunately, we already specified a SecuritySchema that requires a password on the same endpoint.

In this scenario, what is the best way to either have a optional SecuritySchema or another way of letting request ommit it when its not required?

**Comment by sunli829** [comment]
I'm working on this 🙂

**Comment by sunli829** [comment, marked as answer]
Released in `poem-openapi v3.0.3`

https://github.com/poem-web/poem/blob/cc5d458ece1811df08ac9630a0466181ba9f2b34/poem-openapi/tests/security_scheme.rs#L589

**Reply by NexRX** [reply]
Wow that is awesome, signs of a mature web-framework for sure! 😍 Thanks sunli!
